File: app/src/main/java/com/example/engineeringthesis/CategoryActivity.java
package com.example.engineeringthesis;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import com.example.engineeringthesis.Adapter.CategoryAdapter;
import com.example.engineeringthesis.Adapter.SelectLanguageAdapter;

public class CategoryActivity extends AppCompatActivity {
    ListView lv;
    String categories[] = {"Sport","Elektronika","Muzyka"};
    Integer categoriesIcon[]={R.drawable.pilka,R.drawable.elektronika,R.drawable.muzyka};
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_category);

        lv = findViewById(R.id.selectCategoryListView);
        CategoryAdapter selectCategoryAdapter = new CategoryAdapter(CategoryActivity.this,categories,categoriesIcon);
        lv.setAdapter(selectCategoryAdapter);
        lv.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l)
            {
                Intent intent = new Intent(CategoryActivity.this,GameActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/engineeringthesis/CategoryActivity.kt
package com.example.engineeringthesis

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.engineeringthesis.adapter.CategoryTeacherAdapter
import com.example.engineeringthesis.model.CategoryTeacher
import com.example.engineeringthesis.model.Teacher
import com.example.engineeringthesis.viewmodel.CategoryTeacherViewModel
import com.example.engineeringthesis.viewmodel.CategoryViewModel
import com.fasterxml.jackson.databind.ObjectMapper
import dagger.android.support.DaggerAppCompatActivity
import javax.inject.Inject

class CategoryActivity : DaggerAppCompatActivity(), CategoryTeacherAdapter.OnCategoryListener {

    lateinit var categoryViewModel: CategoryViewModel
    lateinit var categoryTeacherViewModel: CategoryTeacherViewModel
    lateinit var recyclerView: RecyclerView
   // lateinit var categoryAdapter:CategoryAdapter
    lateinit var categoryAdapter:CategoryTeacherAdapter
    @Inject
    lateinit var sharedPreferences: SharedPreferences;
    @Inject
    lateinit var sharedPreferencesEditor: SharedPreferences.Editor;
    @Inject
    lateinit var jacksonMapper:ObjectMapper
    var categoriesIcon = arrayOf(R.drawable.pilka, R.drawable.elektronika, R.drawable.muzyka)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_category)
        categoryTeacherViewModel = ViewModelProvider(this).get(CategoryTeacherViewModel::class.java)
        buildRecyclerView()
        var teacherReturned: String =  sharedPreferences.getString("teacher_selected","null").toString()
        var teacher:Teacher = jacksonMapper.readValue(teacherReturned,Teacher::class.java)
        categoryTeacherViewModel.getAllCategoriesByTeacher(teacher.teacherId).observe(this, { categories -> categoryAdapter.setCategoryNamesList(categories) })

    }

    fun buildRecyclerView()
    {
        recyclerView = findViewById(R.id.categoryRecyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.setHasFixedSize(true)
        categoryAdapter = CategoryTeacherAdapter(this)
        recyclerView.adapter = categoryAdapter
    }

    override fun onCategoryClick(position: Int) {
        var cat: CategoryTeacher = categoryAdapter.getItem(position);
        var JsonObject = jacksonMapper.writeValueAsString(cat)
        sharedPreferencesEditor.putString("category_selected",JsonObject).apply()
        val intent = Intent(this, GameActivity::class.java)
        startActivity(intent)

        //Toast.makeText(this, "Kategoria = " + cat.toString() , Toast.LENGTH_LONG).show()
    }
}